distribution/lib/Standard/Database/0.0.0-dev/src/Data/Dialect.enso

```diff
@@ -87,6 +88,40 @@ type Dialect
     get_naming_helpers self =
         Unimplemented.throw "This is an interface only."
 
+    ## PRIVATE
+       Returns the mapping between SQL types of this dialect and Enso
+       `Value_Type`.
+    get_type_mapping : SQL_Type_Mapping
+    get_type_mapping self =
+        Unimplemented.throw "This is an interface only."
```

Comment:
There is `Meta..Type.get_type_methods` since
- #5956 

now we can write a test that compares the methods of an implementation with the methods of the _"interface"_. Shall we start writing such tests as part of this PR?

Reply:
I'm not sure really.

Ideally, we wanted to have an `interface` concept built into the language - that would be great and could do some of this check for us.

I'm not sure if here it is worth testing it that way, because it is tested by all the other tests - we just _use_ the interface implementations, so if the tests pass it is OK - we rely more on "integration tests" of these interfaces.

Reply:
We are not going to have `interface` concept in the language any time soon. The usual substitute for missing _"typing"_ capabilities of a language is to write tests (some said that in Ruby one had to write three times more tests than in Java to get the same quality of code). Special test using the `get_type_methods` would be a general purpose replacement of the missing `interface` concept. If you think other tests substitute that special test, then OK.

Reply:
After some thought, I guess that if we can have a simple test which is just `Test.assume_implements_interface instance interface`, it could indeed be useful.

I still think it's not that necessary at this moment, as it's tested enough, but if we can have such method and ensure consistency "for free", it could be useful, and should be relatively simple to implement. I'll keep it in mind, but won't add now since there is plenty of more important stuff.